modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/configuration/CETConfigurationFactory.java

```diff
@@ -55,69 +57,196 @@ public class CETConfigurationFactory {
-				CETConfiguration cetConfiguration =
-					ConfigurableUtil.createConfigurable(
-						CETConfiguration.class, properties);
-
-				_cet = _cetManager.addCET(
-					cetConfiguration, companyId,
-					_getExternalReferenceCode(properties));
-
-				if (Objects.equals(
-						_cet.getType(),
-						ClientExtensionEntryConstants.TYPE_THEME_CSS)) {
-
-					_addControlPanelThemeCSSClientExtensionEntryRel(companyId);
+				try {
+					if (_log.isInfoEnabled()) {
+						_log.info(
+							StringBundler.concat(
+								"Adding CET for company ", companyId, " and ",
+								"client extension ", name));
+					}
+
+					_cet = _cetManager.addCET(
+						ConfigurableUtil.createConfigurable(
+							CETConfiguration.class, properties),
+						companyId, _getExternalReferenceCode(properties));
+
+					if (Objects.equals(
+							_cet.getType(),
+							ClientExtensionEntryConstants.TYPE_THEME_CSS)) {
+
+						if (_log.isInfoEnabled()) {
+							_log.info(
+								StringBundler.concat(
+									"Adding control panel theme CSS relation ",
+									"for company ", companyId, " and client ",
+									"extension ", name));
+						}
+
+						_addControlPanelThemeCSSClientExtensionEntryRel(
+							companyId);
+					}
+				}
+				catch (Exception exception) {
+					_log.error(
+						StringBundler.concat(
+							"Unable to add CET for company ", companyId,
+							" and client extension ", name),
+						exception);
+
+					throw exception;
 				}
 			});
 	}
 
 	@Deactivate
 	protected void deactivate() {
+		String name = (String)_properties.get("name");
+
+		if (_log.isInfoEnabled()) {
+			_log.info(
+				"Notified from OSGi about deactivation of client extension " +
+					name);
+		}
+
 		ConfigurationFactoryUtil.executeAsCompany(
 			_companyLocalService, _properties,
 			companyId -> {
-				_cetManager.deleteCET(_cet);
+				try {
+					if (_log.isInfoEnabled()) {
+						_log.info(
+							StringBundler.concat(
+								"Deleting CET for company ", companyId, " and ",
+								"client extension ", name));
+					}
+
+					_cetManager.deleteCET(_cet);
+
+					if (_log.isInfoEnabled()) {
+						_log.info(
+							StringBundler.concat(
+								"Deleting client extension relations (if any) ",
+								"for company ", companyId, " and client ",
+								"extension ", name));
+					}
 
-				_clientExtensionEntryRelLocalService.
-					deleteClientExtensionEntryRels(
-						companyId, _cet.getExternalReferenceCode());
+					_clientExtensionEntryRelLocalService.
+						deleteClientExtensionEntryRels(
+							companyId, _cet.getExternalReferenceCode());
+				}
+				catch (Exception exception) {
+					_log.error(
+						StringBundler.concat(
+							"Unable to delete CET for company ", companyId,
+							" and client extension ", name),
+						exception);
+
+					throw exception;
+				}
 			});
 
 		_properties = null;
 	}
 
 	@Modified
 	protected void modified(Map<String, Object> properties) throws Exception {
+		String name = (String)_properties.get("name");
```

Comment:
same as above, take `ERC` here and use for logging

Reply:
Yeah, let's use ERC instead. I thought it could be not meaningful under some circumstances but turns out that that only happens when managing CXs through the UI (when deploying them in .zip files the ERC is given by the developer).